### PR TITLE
[REST API] Merge `WordPressOrgRequest` into `RESTRequest`

### LINKED ISSUE #8512: [REST API] Merge `WordPressOrgRequest` and `RESTRequest`
`WordPressOrgRequest` and `RESTRequest` have almost similar properties and they can be merged together into a single struct.

- Use `PlaceholderDataValidator` as `WordPressApiValidator` is not needed for .org requests.
- Use JSON encoding for `POST` and `PUT` requests

Original discussion [here](https://github.com/woocommerce/woocommerce-ios/pull/8483/files#r1059832482)

### PR DESCRIPTION
**Pull request by selanthiraiyan**
Closes: #8512 

## Description
`WordPressOrgRequest` and `RESTRequest` have similar properties. For clarity we decided to merge them both and use 
 `RESTRequest`. 

## Testing instructions

- We need to test the areas in which `WordPressOrgRequest` (deleted in this PR) and `RESTRequest` are being used.
- Create a test store using JN. (without Jetpack)

*Jetpack installation*
- Log out of the app or skip login onboarding if needed.
- On the prologue screen, select "Log In" or "Continue with WordPress.com" based on the A/B test variant you get.
- Log in with your WordPress.com account.
- If your account doesn't have any associated site, notice that the empty store picker is displayed with the matching design as on Figma.
- Select Add a Store > Connect an existing site.
- Enter the address of your test store and tap Continue.
- You should see a Jetpack error screen from where you can Install Jetpack or Connect Jetpack natively.
- Validate that the Jetpack installation/setup works as before.

*Application password*
1. Turn on the `applicationPasswordAuthenticationForSiteCredentialLogin` feature flag.
1. Login into the app using .org site credentials.
1. Navigate to the coupons tab.
1. Try creating, editing and deleting coupons.
1. All features in the coupons screen should work as before.

## Screenshots
NA

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: Networking/NetworkingTests/Remote/RemoteTests.swift [hunk 1]
```diff
@@ -423,28 +423,32 @@ final class RemoteTests: XCTestCase {
         XCTAssertNotNil(result)
     }
 
-    /// Verifies that WordPressOrg request parses WordPressApiError
+    /// Verifies that RESTRequest request doesn't parse WordPressApiError
     ///
-    func test_wordpress_org_request_parses_wordpress_api_error() async throws {
+    func test_wordpress_org_request_does_not_parse_wordpress_api_error() async throws {
```
Comment (itsmeichigo) [inline Networking/NetworkingTests/Remote/RemoteTests.swift:428]: I don't understand this test case, should the request parse the error though?

File: Networking/NetworkingTests/Remote/RemoteTests.swift [hunk 2]
```diff
@@ -423,28 +423,32 @@ final class RemoteTests: XCTestCase {
...
 
-        network.simulateResponse(requestUrlSuffix: "mock", filename: "error-wp-rest-forbidden")
+        network.simulateResponse(requestUrlSuffix: "mock", filename: "timeout_error")
 
-        await assertThrowsError({ _ = try await remote.enqueue(request, mapper: mapper)}, errorAssert: { $0 is WordPressApiError })
+        // When
+        let result = try await remote.enqueue(request, mapper: mapper)
+
+        // Then
+        XCTAssertNotNil(result)
```
Comment (itsmeichigo) [inline Networking/NetworkingTests/Remote/RemoteTests.swift:441]: Should we check that `result` is `failure` because we are mocking a timeout error?

**Review by itsmeichigo — APPROVED** [review]
LGTM, I left a question regarding the tests.

File: Networking/NetworkingTests/Remote/RemoteTests.swift (hunk 1 above)
Comment (selanthiraiyan) [reply Networking/NetworkingTests/Remote/RemoteTests.swift:428]: The original test case validated that `WordPressApiValidator` is used in `WordPressOrgRequest`. 

In `RESTRequest` we decided to use `PlaceholderDataValidator` which will not validate the request. So, I updated the test case to check that no errors are parsed. 
i.e. `test_wordpress_org_request_does_not_parse_wordpress_api_error` and `test_wordpress_org_request_does_not_parse_dotcom_error` will ensure that `PlaceholderDataValidator` is used.

File: Networking/NetworkingTests/Remote/RemoteTests.swift (hunk 2 above)
Comment (selanthiraiyan) [reply Networking/NetworkingTests/Remote/RemoteTests.swift:441]: I followed this pattern from the `test_wordpress_org_request_does_not_parse_dotcom_error` method.

The idea here is ensure `try await remote.enqueue(request, mapper: mapper)` doesn't throw any error. 

Please note that the `result` is of type `Data` from `DummyMapper`, and not of `Swift.Result<Data, Error>` type.

**Comment by selanthiraiyan** [comment]
Thank you for the review, Huong! 🙇 

I am going ahead and merging this now. I will make changes to the unit tests if anything comes up from our PR review discussions.

**Pull request merged**